packages/app/features/account/sendtag/checkout/checkout-form.tsx

```diff
@@ -363,145 +64,238 @@ function ReferredBy() {
     },
   })
   const { data: referrer, error: referrerError } = useReferrer()
+  const { data: referred, isLoading: isLoadingReferred } = api.referrals.getReferred.useQuery()
+  const [isInputFocused, setIsInputFocused] = useState<boolean>(false)
 
-  return (
-    <YStack w="100%" mt="$4" gap="$2" borderBottomWidth={1} pb="$6" borderColor="$decay">
-      <Paragraph
-        fontFamily={'$mono'}
-        fontWeight={'400'}
-        $theme-light={{ col: '$gray11Light' }}
-        $theme-dark={{ col: '$gray11Dark' }}
-        fontSize={'$4'}
-        mb="$0"
-        pb="$0"
-      >
-        Referred by someone? Enter their referral code below. No slash before.
-      </Paragraph>
-      <XStack gap="$2" ai={'center'} jc="flex-start">
-        <YStack jc="flex-start" ai="flex-start">
-          <Label fontWeight="500" col={'$color12'} htmlFor={'refcode'}>
-            Referral Code:
-          </Label>
-          <XStack gap="$2" jc="flex-start" ai="flex-start">
+  switch (true) {
+    case isLoadingReferred:
+      return <Spinner color="$color11" />
+    case !!referred:
+      return null
+    default:
```

Comment:
While I love pattern matching, `switch (true)` is not quite there. Should try to avoid it if we can. I only use it to avoid nested ternaries

instead let's use some early returns
```suggestion
if (isLoadingReferred)    return <Spinner color="$color11" />
if (!!referred) return null
```